### PR TITLE
Some fixes for fclayers_hooks

### PR DESCRIPTION
**Pull request by Koncopd**

### CODE REVIEW

File: scvi/model/scanvi.py [hunk 1]
```diff
@@ -259,6 +259,10 @@ def train(
         self.trainer.labelled_set = self.trainer.create_scvi_dl(
             indices=self._labeled_indices
         )
+
+        for scdl in [self.trainer.labelled_set, self.trainer.unlabelled_set]:
+            scdl.to_monitor = ["reconstruction_error", "accuracy"]
+
```
Comment (adamgayoso) [inline scvi/model/scanvi.py:265]: isn't this already done?

https://github.com/YosefLab/scvi-tools/blob/b519575a5469f1ec71250e06a8856e155bbf5c91/scvi/core/trainers/annotation.py#L187-L188

File: scvi/core/modules/scanvae.py [hunk 2]
```diff
@@ -176,8 +176,8 @@ def classify(self, x, batch_index=None):
             w_y = self.classifier(z)
         return w_y
 
-    def get_latents(self, x, y=None):
-        zs = super().get_latents(x)
+    def get_latents(self, x, y=None, batch_index=None):
```
Comment (adamgayoso) [inline scvi/core/modules/scanvae.py:179]: These functions are not actually used anywhere, we could also consider removing them now.

File: scvi/model/scanvi.py (hunk 1 above)
Comment (Koncopd) [reply scvi/model/scanvi.py:265]: The problem is that it is done in trainer and then scanvi assigns datasets after trainer init without `to_monitor`.

File: scvi/core/modules/scanvae.py (hunk 2 above)
Comment (Koncopd) [reply scvi/core/modules/scanvae.py:179]: We use them, but they can be replaced ofc.

File: scvi/model/scanvi.py (hunk 1 above)
Comment (adamgayoso) [reply scvi/model/scanvi.py:265]: lol, we really need to refactor all the semisupervised trainers

File: scvi/core/modules/scanvae.py (hunk 2 above)
Comment (adamgayoso) [reply scvi/core/modules/scanvae.py:179]: are you guys able to use instead the API as shown in the colab notebooks I linked in the PR?

**Pull request closed without merging**